src/http/server/handlers/static_file_handler.cr

```diff
@@ -76,6 +80,19 @@ class HTTP::StaticFileHandler
       end
 
       context.response.content_type = MIME.from_filename(file_path.to_s, "application/octet-stream")
+
+      # Checks if pre-gzipped file can be served
+      gz_file_path = "#{file_path}.gz"
+      if File.exists?(gz_file_path) &&
+         # Allow small time drift. In some file systems, using `gz --keep` to
+         # compress the file will keep the modification time of the original file
+         # but truncating some decimals
+         last_modified - modification_time(gz_file_path) < 1.millisecond &&
+         context.request.headers.includes_word?("Accept-Encoding", "gzip")
```

Comment:
Maybe we can check this condition before checking that the file exists? That way we can potentially avoid a syscall.

Reply:
Really good point